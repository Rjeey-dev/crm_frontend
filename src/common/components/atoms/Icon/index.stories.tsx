import React from 'react';
import { storiesOf } from '@storybook/react';
import Icon from './index';

storiesOf('Icon', module)
    .add('notifications', () => (
        <Icon classes='mdi mdi-file-document-box'/>
    ))
    .add('settings', () => (
        <Icon classes='mdi mdi-tune'/>
    ));