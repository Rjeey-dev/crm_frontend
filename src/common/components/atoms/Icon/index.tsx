import React from 'react';

interface IProps {
    classes: string,
    attributes?: object
}

function Icon(props: IProps) {
    return <i className={props.classes} {...props.attributes}/>;
}

export default Icon;
