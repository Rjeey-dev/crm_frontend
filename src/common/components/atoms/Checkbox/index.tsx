import React, {Component, ReactNode, ReactNodeArray} from "react";

import Icon from 'common/components/atoms/Icon/index';
import Label from "common/components/atoms/Label/index";

interface IProps {
    classes?: string,
    children?: ReactNodeArray | ReactNode,
    checked?: boolean,
    disabled?: boolean,
    onChange: (data: any) => void,
}

class Checkbox extends Component<IProps> {
    public state = {
        checked: false
    };

    componentWillReceiveProps(props: IProps): void {
        if (props.checked) {
            this.setState({
                checked: props.checked
            })
        }
    }

    componentDidMount(): void {
        if (this.props.checked) {
            this.setState({
                checked: this.props.checked
            })
        }
    }

    render() {
        const disabled = this.props.disabled ? this.props.disabled : false;

        return <div className="form-check form-check-flat mt-0">
            <Label classes="form-check-label">
                <input type="checkbox" disabled={disabled} onChange={() => {
                    this.setState({
                        checked: !this.state.checked
                    }, () => {
                        this.props.onChange(this.state.checked)
                    })
                }}
                       className={this.props.classes}
                       checked={this.state.checked}/>{this.props.children}<Icon classes="input-helper"/>
            </Label>
        </div>;
    }
}

export default Checkbox;

