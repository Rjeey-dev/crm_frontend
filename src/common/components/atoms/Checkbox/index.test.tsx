import { shallow } from 'enzyme';
import React from 'react';

import Checkbox from './index';
import Icon from "../Icon/index";

it("Renders <Checkbox/> with correct properties", () => {
    const output = shallow(
        <Checkbox classes='class'>
            <Icon classes='icon'/>
        </Checkbox>
    ).html();

    expect(output).toMatchSnapshot();
});

it("Renders <Checkbox/> with only required props", () => {
    const output = shallow(
        <Checkbox classes='class'/>
    ).html();

    expect(output).toMatchSnapshot();
});