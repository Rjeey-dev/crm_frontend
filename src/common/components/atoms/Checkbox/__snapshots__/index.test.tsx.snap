// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders <Checkbox/> with correct properties 1`] = `"<div class=\\"form-check form-check-flat mt-0\\"><label class=\\"form-check-label\\"><input type=\\"checkbox\\" class=\\"class\\" checked=\\"\\"/><i class=\\"icon\\"></i><i class=\\"input-helper\\"></i></label></div>"`;

exports[`Renders <Checkbox/> with only required props 1`] = `"<div class=\\"form-check form-check-flat mt-0\\"><label class=\\"form-check-label\\"><input type=\\"checkbox\\" class=\\"class\\" checked=\\"\\"/><i class=\\"input-helper\\"></i></label></div>"`;
